src/common/utility.c

```diff
@@ -294,6 +294,22 @@ gboolean dt_util_test_image_file(const char *filename)
   return regular && size_ok;
 }
 
+gboolean dt_util_test_writable_dir(const char *path)
+{
+  if(path == NULL) return FALSE;
+#ifdef _WIN32
+  struct _stati64 stats;
+  if(_stati64(path, &stats)) return FALSE;
+#else
+  struct stat stats;
+  if(stat(path, &stats)) return FALSE;
+#endif
+  if(S_ISDIR(stats.st_mode) == 0) return FALSE;  
+  if(g_access(path, W_OK | X_OK) != 0) return FALSE;
```

Comment:
Can't we avoid a call to g_access() here by check the stat st_mod filed. I quote the Linux man:


```
       The following mask values are defined for the file  mode  component  of
       the st_mode field:

           S_ISUID     04000   set-user-ID bit (see execve(2))
           S_ISGID     02000   set-group-ID bit (see below)
           S_ISVTX     01000   sticky bit (see below)

           S_IRWXU     00700   owner has read, write, and execute permission
           S_IRUSR     00400   owner has read permission
           S_IWUSR     00200   owner has write permission
           S_IXUSR     00100   owner has execute permission

           S_IRWXG     00070   group has read, write, and execute permission
           S_IRGRP     00040   group has read permission
           S_IWGRP     00020   group has write permission
           S_IXGRP     00010   group has execute permission

           S_IRWXO     00007   others  (not  in group) have read, write, and
                               execute permission
           S_IROTH     00004   others have read permission
           S_IWOTH     00002   others have write permission
           S_IXOTH     00001   others have execute permission
```

Reply:
I think that's exactly what @cryptomilk meant. For linux it's that easy, sure. But for windows, i am not so sure. Reading through the docs for g_stat and g_access and references didn't help here, one further reference

https://web.archive.org/web/20101214222704/http://msdn.microsoft.com/en-us/library/14h5k7ff(v=vs.80).aspx

makes me very suspicious about it and so i trusted g_access.

Reply:
Sure, but let's the speed penalty for Windows and so I would include the g_access() inside the Windows branch and do the check of st_mode on Linux.

Reply:
Are we sure we don't have ACL in use? access() would serve them too, stat wouldn't. I always feel pretty uneasy about this stuff, don't care here about performance but would always prefer high level functions.

Reply:
Agreed, I forgot about ACL which are not handled by stat().